### ISSUE TITLE
MacOS (arm64) - VPN does not capture traffic from "Speedtest" App in the App Store.

### ISSUE THREAD

**Issue by PaulCharlton** [mullvad/mullvadvpn-app#5802]
### Is it a bug?

- [X] I know this is an issue with the app, and contacting Mullvad support is not relevant.

### I have checked if others have reported this already

- [X] I have checked the issue tracker to see if others have reported similar issues.

### Current Behavior

MacOS (arm64) - VPN does not capture traffic from "Speedtest" App in the App Store.

Install SpeedTest App; Run App;  Even when VPN is using remote server, SpeedTest still detects and uses regionally local server for its testing.

Without further inspection, I imagine that the SpeedTest app is setting up some sort of in-process bridging to the NIC, or accessing the NIC via a Framework which does not go through the TCP/IP filter used by the VPN stack

### Expected Behavior

MacOS (arm64) - VPN does not capture traffic from "Speedtest" App in the App Store.

Install SpeedTest App; Run App;  Even when VPN is using remote server, SpeedTest should detect and use server regionally local to the VPN designated endpoint. [ie: if I am in London, and my VPN endpoint is is Germany, SpeedTest should use a server close to the German endpoint]

### Steps to Reproduce

Tested on MacOS (arm64) Sonoma

Install SpeedTest App; Run App;  Even when VPN is using remote server, SpeedTest still detects and uses regionally local server for its testing.

### Operating system version

macOS Sonoma (arm64)

### Mullvad VPN app version

2023.6

**Comment by raksooo** [comment]
Thanks for reporting! Which speedtest app are you using? Is it "Speedtest by Ookla"? We tried that one but couldn't reproduce this issue.

Our app uses the packet filter firewall to block traffic from going outside the tunnel. Hopefully it uses some other kind of geolocation method.

**Comment by PaulCharlton** [comment]
yes, "Speed Test by Ookla" -- screen caps attached.  re-verified the behavior.
One thing to note, is that the Speed Test does appear to geo-locate the ISP that the VPN forwards from, but it is still
finding the geographically nearest server to the physical machine -- see screen caps.
Ashburn VA is very close the my physical location at the time of this writing, and it is what comes up when I choose various VPN endpoints around the world.

<img width="266" alt="Screenshot 2024-02-17 at 10 24 20 PM" src="https://github.com/mullvad/mullvadvpn-app/assets/1747363/b8ef1e4c-ef9e-42db-8ef9-3a19d7f998c2">
<img width="311" alt="Screenshot 2024-02-17 at 10 24 07 PM" src="https://github.com/mullvad/mullvadvpn-app/assets/1747363/74d4d7b7-8f1d-4fe2-b173-984f97ac12af">
<img width="384" alt="Screenshot 2024-02-17 at 10 23 50 PM" src="https://github.com/mullvad/mullvadvpn-app/assets/1747363/f5928f35-c482-4e8a-b741-1d9b01f1dd84">
<img width="371" alt="Screenshot 2024-02-17 at 10 23 34 PM" src="https://github.com/mullvad/mullvadvpn-app/assets/1747363/6785eab3-3c20-4f33-b146-54692be13030">
<img width="377" alt="Screenshot 2024-02-17 at 10 22 58 PM" src="https://github.com/mullvad/mullvadvpn-app/assets/1747363/00bb71d7-b57c-4f7b-8133-1784ef784280">
<img width="447" alt="Screenshot 2024-02-17 at 10 22 28 PM" src="https://github.com/mullvad/mullvadvpn-app/assets/1747363/50118eac-e9b6-46c2-9a88-a0e0c6564a8d">

**Comment by Joe12387** [comment]
Hi there.

I downloaded this app and it requests permissions for Location Services. I think the reporter may have granted permissions to geolocation and that may be how the app knows the location.

<img width="276" alt="Screenshot 2024-02-26 at 5 02 37 PM" src="https://github.com/mullvad/mullvadvpn-app/assets/489322/68093828-8cdb-424c-ac53-738ef9f4ef28">

@PaulCharlton Try disabling Location Services from Privacy & Security in Settings to see if this is the issue.

Thanks.

**Comment by MahdiNazemi** [comment]
My experience with the Speedtest website is that it chooses a nearby server based on latency, and switching to VPN servers in other locations does not change the initially selected Speedtest server. I have to hard refresh the page for Speedtest to choose a new server close to the newly selected Mullvad server. I am not sure if the app behaves the same way, but closing and re-opening the app after switching to a new Mullvad server might help.

**Comment by Rawa** [comment]
Closing issue as this seems not to be an app issue, please comment if you want me to reopen and still think this is a problem.
